jfix-zookeeper/src/main/kotlin/ru/fix/zookeeper/discovery/ServiceDiscovery.kt

```diff
@@ -0,0 +1,80 @@
+package ru.fix.zookeeper.discovery
+
+import org.apache.curator.framework.CuratorFramework
+import org.apache.zookeeper.CreateMode
+import org.apache.zookeeper.KeeperException
+import org.slf4j.LoggerFactory
+import ru.fix.zookeeper.transactional.TransactionalClient
+import ru.fix.zookeeper.utils.Marshaller
+import ru.fix.zookeeper.utils.ZkTreePrinter
+
+/**
+ * This class provides functionality of instance id management.
+ * When this class instantiated, it generates new instance id for this zookeeper client and registers this id in znode.
+ * When zookeeper client died or connection closed, znode with generated instance id also deleted.
+ * Service registration guarantees uniqueness of instance id in case of parallel startup on different JVMs.
+ */
+class ServiceDiscovery(
```

Comment:
ServiceDiscovery is a well know pattern. Here we doing something else: this class actually `...provides functionality of instance id management`. So let's name this class `ServiceInstanceIdRegistry`

Reply:
renamed